tools/monorepo-utils/src/github/repo.ts

```diff
@@ -0,0 +1,42 @@
+/**
+ * External dependencies
+ */
+import { gt as greaterThan } from 'semver';
+import { Repository } from '@octokit/graphql-schema';
+
+/**
+ * Internal dependencies
+ */
+import { graphqlWithAuth } from './api';
+
+export const getLatestReleaseVersion = async ( options: {
+	owner?: string;
+	name?: string;
+} ): Promise< string > => {
+	const { owner, name } = options;
+
+	const data = await graphqlWithAuth< { repository: Repository } >( `
+			{
+			    repository(owner: "${ owner }", name: "${ name }") {
+					releases(
+						first: 25
+						orderBy: { field: CREATED_AT, direction: DESC }
+					) {
+						nodes {
+							tagName
+						}
+					}
+				}
+			}
+		` );
+
+	return data.repository.releases.nodes
+		.map( ( node ) => node.tagName )
+		.filter( ( tagName ) =>
+			// Remove any non-numeric and decimal tags, ie prerelease, nightly, and beta tester tags.
+			/^[\d.]+$/.test( tagName )
```

Comment:
>Can't you just find the first release that has the isLatest field?

Thanks for the suggestion @samueljseay. Looking at the data on `woocommerce/woocommerce` with the `isLatest` field supplied appears that it will work well.

<img width="417" alt="image" src="https://user-images.githubusercontent.com/1922453/234130220-598336c8-d284-4622-8e7a-5df5bc62d4a6.png">

However, what if the Beta Tester was the last release made from the monorepo? Then `isLatest` would incorrectly surface the wrong release.

>I think otherwise if we release something for another package like we do with beta tester and it has a higher version number it'll be returned instead. Your test case in jest looks like maybe the only reason it doesn't return wc-beta-tester is that 99.99.00 is not a valid version number?

In this case the regex you commented on, `/^[\d.]+$/.test( tagName )`, tests for the presence of only digits, so releases like `nightly`, `wc-beta-tester-99.99.00` or `wc-beta-tester-99.99.0` will be filtered out and only WooCommerce Core releases will be compared.

I fixed the test sample data to have a valid version for the Beta Tester in https://github.com/woocommerce/woocommerce/pull/37723/commits/d4e8dcdddb3fb71c41ce53b4a75355e8a8ea9f30

Reply:
Fixed in https://github.com/woocommerce/woocommerce/pull/37723/commits/9571f0b126afdf101f10f932c24c513697baec59.

Beta Tester and prereleases never get the `isLatest: true` property